include/circt/Dialect/Calyx/CalyxOps.h

```diff
@@ -37,14 +37,35 @@ class ControlLike : public mlir::OpTrait::TraitBase<ConcreteType, ControlLike> {
   }
 };
 
-/// The direction of a Calyx port.
-enum PortDirection { INPUT = 0, OUTPUT = 1 };
+/// The port direction attribute follows the implementation style of FIRRTL
+/// module port direction attributes.
+enum Direction { Input = 0, Output };
+namespace direction {
+
+/// The key in a components's attribute dictionary used to find the direction.
+constexpr const char *attrKey = "portDirections";
+
+/// Return an output direction if \p isOutput is true, otherwise return an
+/// input direction.
+Direction get(bool isOutput);
+
+/// Return a \p IntegerAttr containing the packed representation of an array
+/// of directions.
+IntegerAttr packAttribute(ArrayRef<Direction> a, MLIRContext *b);
```

Comment:
Uber nit: but if you're going to name the parameters in the header here, I'd recommend using the more descriptive names that you gave them in the cpp file rather than just `a` and `b`.

Reply:
As above.